### PR TITLE
Add support for ALC (Angelcoin)

### PR DESCRIPTION
**Pull request by angelcoinmoney**

### CODE REVIEW

File: vanitygen.c [hunk 1]
```diff
@@ -1386,6 +1406,14 @@ main(int argc, char **argv)
 			addrtype = atoi(optarg);
 			privtype = 128 + addrtype;
 			scriptaddrtype = addrtype;
+			//Angelcoin Modification
```
Comment (exploitagency) [inline vanitygen.c]: Start block not needed

File: vanitygen.c [hunk 2]
```diff
@@ -1386,6 +1406,14 @@ main(int argc, char **argv)
...
+			if ( addrtype == 23 )
+			//ALC [Angelcoin] Address
+				scriptaddrtype = 0;
+			else
+			if ( addrtype == 83 )
+			//ALC [Angelcoin] Testnet Address
+				scriptaddrtype = 3;
```
Comment (exploitagency) [inline vanitygen.c]: End block

File: keyconv.c [hunk 3]
```diff
@@ -1064,7 +1082,7 @@ main(int argc, char **argv)
 			break;
 		case 'X':
 			addrtype_opt = atoi(optarg);
-			privtype_opt = addrtype + 128;
+			privtype_opt = addrtype_opt + 128;
```
Comment (exploitagency) [inline keyconv.c]: I think you have found a bug, I don't know how it has been like this for so long and slipped through the cracks, I suppose no one uses this feature due to arg `-C`.

File: keyconv.c [hunk 4]
```diff
@@ -195,6 +197,22 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "ALC")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Address\n");
+					addrtype_opt = 23;
+					privtype_opt = 23+128;
```
Comment (exploitagency) [inline keyconv.c]: Why don't you just add them together?

File: keyconv.c [hunk 5]
```diff
@@ -88,6 +88,8 @@ main(int argc, char **argv)
 					"Argument(UPPERCASE) : Coin : Address Prefix\n"
 					"---------------\n"
 					"42 : 42coin : 4\n"
+					"ALC : Angelcoin : A\n"
```
Comment (exploitagency) [inline keyconv.c]: Please arrange in alphabetical order

File: README.md [hunk 6]
```diff
@@ -134,6 +134,8 @@ Current List of Available Coins for Address Generation
 |**Argument(UPPERCASE)** | **Coin** | **Address Prefix**  |
 | --------------------------------------- | -------------------------------------------- | ------------ |
 |42 | 42coin | 4  |
+|ALC | Angelcoin | A  |
```
Comment (exploitagency) [inline README.md]: Please arrange in alphabetical order

File: oclvanitygen.c [hunk 7]
```diff
@@ -183,6 +183,8 @@ main(int argc, char **argv)
 					"Argument(UPPERCASE) : Coin : Address Prefix\n"
 					"---------------\n"
 					"42 : 42coin : 4\n"
+					"ALC : Angelcoin : A\n"
```
Comment (exploitagency) [inline oclvanitygen.c]: Please arrange in alphabetical order

File: vanitygen.c [hunk 8]
```diff
@@ -437,6 +437,8 @@ main(int argc, char **argv)
 					"Argument(UPPERCASE) : Coin : Address Prefix\n"
 					"---------------\n"
 					"42 : 42coin : 4\n"
+					"ALC : Angelcoin : A\n"
```
Comment (exploitagency) [inline vanitygen.c]: Please arrange in alphabetical order

File: vanitygen.c [hunk 9]
```diff
@@ -544,6 +546,24 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "ALC")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Address\n");
+					addrtype = 23;
+					privtype = 23+128;
```
Comment (exploitagency) [inline vanitygen.c]: Add

File: vanitygen.c [hunk 10]
```diff
@@ -544,6 +546,24 @@ main(int argc, char **argv)
...
+					scriptaddrtype = 0;
+					break;
+			}
+			else
+			if (strcmp(optarg, "ALCtest")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Testnet Address\n");
+					addrtype = 83;
+					privtype = 83+128;
```
Comment (exploitagency) [inline vanitygen.c]: Add

File: oclvanitygen.c [hunk 11]
```diff
@@ -289,6 +291,22 @@ main(int argc, char **argv)
...
+					addrtype = 23;
+					privtype = 23+128;
+					break;
+			}
+			else
+			if (strcmp(optarg, "ALCtest")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Testnet Address\n");
+					addrtype = 83;
+					privtype = 83+128;
```
Comment (exploitagency) [inline oclvanitygen.c]: Add

File: oclvanitygen.c [hunk 12]
```diff
@@ -289,6 +291,22 @@ main(int argc, char **argv)
 					return 1;
 			}
 			else
+			if (strcmp(optarg, "ALC")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Address\n");
+					addrtype = 23;
+					privtype = 23+128;
```
Comment (exploitagency) [inline oclvanitygen.c]: Add

File: keyconv.c [hunk 13]
```diff
@@ -195,6 +197,22 @@ main(int argc, char **argv)
...
+					addrtype_opt = 23;
+					privtype_opt = 23+128;
+					break;
+			}
+			else
+			if (strcmp(optarg, "ALCtest")== 0) {
+				fprintf(stderr,
+					"Generating ALC [Angelcoin] Testnet Address\n");
+					addrtype_opt = 83;
+					privtype_opt = 83+128;
```
Comment (exploitagency) [inline keyconv.c]: Add

File: keyconv.c (hunk 3 above)
Comment (exploitagency) [reply keyconv.c]: Please remove this line, have you even compiled and tested the application?

**Comment by AngelTs** [comment]
I made the changes! What else to do?

**Comment by angelcoinmoney** [comment]
I made the request changes! What else to do?

**Comment by exploitagency** [comment]
Sorry I have not been able to review it again yet, and I read the notification when I didnt have time to look it over so your pr got a bit lost, thats my fault. Will try this weekend.

**Pull request closed without merging**